### DISCUSSION TITLE
Dismissing the slideshow on swipe down

### DISCUSSION THREAD

**Discussion by nk9** [igordanchenko/yet-another-react-lightbox#125] [Ideas]
I'd like to let mobile users dismiss the lightbox with a swipe down gesture. This is a common mechanism in iOS.

Is there a way to receive a callback for the swipe down gesture event when the lightbox is active?

**Comment by igordanchenko** [comment]
Thank you for bringing this up. I had previously considered implementing this feature, but ultimately chose to postpone it because, in my opinion, it should be accompanied by visually appealing animation and a seamless UX.

If you are referring to the Photos app on iOS, it offers a highly user-friendly UX with the following features:

1. The photo moves with your finger as you pull it down.
2. The background gradually fades out as you pull it down further.
3. The image snaps back into its thumbnail position once you release your finger.

I would prefer to have a somewhat similar UX in the lightbox, but I anticipate that the implementation might be tedious.

Regardless, I have drafted this feature without any visual effects for now, and you can test it in the following sandbox.

https://8fjq5t-5173.csb.app

Please let me know your thoughts. If you like what you see, I can release this feature as-is and implement the visual effects sometime in the future.

**Reply by nk9** [reply]
Thanks very much for the reply and the quick work on this!

I definitely agree that including the smooth dismiss UX is ideal as a long-term solution. Of the three UX elements you suggest, though, I think the first is by far the most important. Yes, I'm thinking of Photos, but other apps do this different ways, and the only consistent thing is that the photo moves along with your finger. Here are Twitter and Toot:

https://github.com/igordanchenko/yet-another-react-lightbox/assets/3646730/12e22fbc-e44e-4b94-b849-c48e06b5b2dd

https://github.com/igordanchenko/yet-another-react-lightbox/assets/3646730/11b7cdaa-5f5a-4b6f-ad7e-47767692ea4f

I don't have a sense for how difficult the three elements are. I do think that it's better to have a gesture which works than one which looks slick, but I'm also personally not in a huge hurry. If you think item 1 alone is straightforward, I'd be happy to save you the hassle of producing a release and wait a bit for that.

But yes, if it's complex and you don't have time now, I'll gladly take the un-UXed gesture now.

**Reply by igordanchenko** [reply]
Thank you for your feedback and the additional examples. This is very helpful.

I went ahead and released the current version of this feature in v3.7.0 (there is really no hassle in pushing the release out - it's fully automated).

```jsx
<Lightbox
  controller={{ closeOnPullDown: true }}
  // ...
/>
```

I'll update this thread when I have an update on the visual effects implementation.

**Reply by igordanchenko** [reply]
I've implemented the pull-down animation in v3.8.2. Please let me know if you have any feedback.

**Reply by nk9** [reply]
Thanks very much for this! I am using it now and it's working well.
